### PR TITLE
add project readme

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Standard updated with the new upload files & logs scripts steps

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,232 @@
...
+ - Docker
+ - Bash
+ 
+Development requires the following additional tools:
+ - Python 3.6
+ - pipenv
+ - git
+
+## Usage
+A pipeline run consists of the following five steps, executed in sequence:
```
Comment (as2388) [inline README.md]:
```suggestion
A pipeline run consists of the following steps, executed in sequence:
```

File: README.md [hunk 2]
```diff
@@ -1 +1,232 @@
...
+1. Download coded data from Coda.
+2. Fetch all the relevant data from Rapid Pro.
+3. Process the raw data to produce the outputs required for coding and then for analysis.
+4. Upload the new data to Coda for manual verification and coding.
+5. Run automated analysis.
+6. Back-up the project data root.
+7. Upload execution logs/archives to long term storage.
```
Comment (as2388) [inline README.md]: Update to match the new run scripts

File: README.md [hunk 3]
```diff
@@ -1 +1,232 @@
...
+
+## Pre-requisites
+Before the pipeline can be run, the following tools must be installed:
+ - Docker
+ - Bash
+ 
+Development requires the following additional tools:
+ - Python 3.6
+ - pipenv
+ - git
```
Comment (as2388) [inline README.md:15]: Add a section that explains that you now need to pipenv --three sync after cloning

**Review by as2388 — APPROVED** [review]
Thanks for bringing this up to date, have added some suggestions for a few other things we should update

File: README.md (hunk 1 above)
Comment (IsaackMwenda) [reply README.md]: updated thanks.

File: README.md (hunk 2 above)
Comment (IsaackMwenda) [reply README.md]: updated thanks.

File: README.md (hunk 3 above)
Comment (IsaackMwenda) [reply README.md:15]: updated thanks.

Comment (IsaackMwenda) [reply README.md:15]: @as2388  Please take another quick look.

File: README.md [hunk 4]
```diff
@@ -1 +1,236 @@
...
+ 
+After cloning you need to run `pipenv sync` in the project directory to install all dependencies required by the last step
```
Comment (as2388) [inline README.md]:
```suggestion
After cloning run `$ pipenv sync` in the project directory to install all dependencies required by the last step 
```

File: README.md [hunk 5]
```diff
@@ -1 +1,236 @@
...
+where:
+- `user` is the identifier of the person running the script, for use in the TracedData Metadata 
+  e.g. `user@africasvoices.org`.
+- `google-cloud-credentials-file-path` is an absolute path to a json file containing the private key credentials
+  for accessing a cloud storage credentials bucket containing all the other project credentials files.
+- `pipeline-configuration-file-path` is an absolute path to a pipeline configuration json file.
+- `run-id` is a unique identifier for the run being uploaded. This will be included in all of the uploaded file names.
+- `data-root` is an absolute path to the directory in which all pipeline data should be stored.
+
+### 8. Upload logs files
```
Comment (as2388) [inline README.md]:
```suggestion
### 8. Upload log files
```

File: README.md [hunk 6]
```diff
@@ -1 +1,236 @@
...
+This stage uploads the archive produced in the backup-data stage and a memory profile log of the generate outputs stage
+to project google drive directory.
```
Comment (as2388) [inline README.md]:
```suggestion
to google cloud storage.
```

File: README.md [hunk 7]
```diff
@@ -1 +1,236 @@
...
+```
+
+where:
+- `data-root` is an absolute path to the directory to back-up.
+- `data-backups-dir` is a directory which the `data-root` directory will be backed-up to.
+  The data is gzipped and given the name `data-<utc-date-time-now>-<git-HEAD-hash>`.
+
+
+### 7. Upload analysis files
+This stage uploads the analysis files generated by generate outputs & automated analysis stage to Google Cloud storage.
```
Comment (as2388) [inline README.md]:
```suggestion
This stage uploads the analysis files generated by generate outputs & automated analysis stage to Google Drive.
```

**Review by as2388 — APPROVED** [review]
Thanks for bringing this up to date

**Review by as2388 — APPROVED** [review]
Thanks for bringing this up to date

**Pull request merged**